desk

```diff
@@ -138,7 +141,11 @@ cmd_go() {
 cmd_run() {
     local TODESK="$1"
     shift;
-    cmd_go "$TODESK" -ic "$@"
+    if [ $# -eq 1 ]; then
+        cmd_go "$TODESK" -ic "$1"
+    else
+        cmd_go "$TODESK" -ic '"$@"' -- "$@"
```

Comment:
hm, shouldn't this be `-ic '"$1"' -- "${@:2}"` or something along those lines? I haven't looked at this code in a while and so could be totally wrong. :sweat_smile:

Reply:
Thanks for looking at this.

The first `'"$@"'` (escaped) is evaluated by the desk shell and expands to the arguments passed after `--`.
The second `"$@"` is passing the full argument vector to the desk shell.

It looks a little odd to just run `"$@"`, but it's a useful idiom for shell middleware.